### PR TITLE
Fix #478 by implementing al_set_new_window_title() and al_get_new_window_title()

### LINKED ISSUE #478: Add al_set_new_window_title
Like it says above. Avoids an visually annoying split second wrong title.

### PR DESCRIPTION
**Pull request by beoran**
Fix #478. 
Also adds an example for the two functions and for al_set_window_tittle().

### CODE REVIEW

File: examples/ex_window_title.c [hunk 1]
```diff
@@ -0,0 +1,112 @@
+/* An example showing bitmap flipping flags, by Steven Wallace. */
```
Comment (SiegeLord) [inline examples/ex_window_title.c]: This comment is wrong.

File: include/allegro5/display.h [hunk 2]
```diff
@@ -110,6 +110,9 @@ AL_FUNC(void, al_set_new_display_flags, (int flags));
 AL_FUNC(int,  al_get_new_display_refresh_rate, (void));
 AL_FUNC(int,  al_get_new_display_flags, (void));
 
+AL_FUNC(void, al_set_new_window_title, (char * title));
```
Comment (SiegeLord) [inline include/allegro5/display.h]: Extra space after the `char *`

File: src/tls.c [hunk 3]
```diff
@@ -181,6 +185,43 @@ ALLEGRO_EXTRA_DISPLAY_SETTINGS *_al_get_new_display_settings(void)
...
+
+
+
+/* Function: al_get_new_window_title
+ */
+const char * al_get_new_window_title(void)
+{
+   thread_local_state *tls;
+   
+   /*Return al_get_prgram name in case of error or if not set before. */
```
Comment (SiegeLord) [inline src/tls.c]: `al_get_prgram` -> `al_get_app_name`

**Comment by beoran** [comment]
With the cleanups above, hopefully I have fixed the points raised in SiegeLord's comments.

**Comment by beoran** [comment]
@SiegeLord is it OK to merge now?

**Comment by SiegeLord** [comment]
It occurs to me that this new string is never freed. I don't know if we leak other things like that, but perhaps we shouldn't. If you want, you could add a `_al_tls_shutdown` function to tls.c and free/NULL it when `al_uninstall_system` is called. Otherwise, it looks good to me.

**Comment by beoran** [comment]
That's a good idea. I'll implement a way to clean up that string.

**Comment by beoran** [comment]
@SiegeLord, there is another snag, al_restore_state and al_save_state, with the api they have cannot be used to manage allocated memory because there is no al_destroy_state. I'd say we just punt an use a static buffer instead, and document the limit of the buffer.

**Comment by SiegeLord** [comment]
Ah, yikes. Yeah, a static buffer seems to be a good solution.

**Comment by beoran** [comment]
@SiegeLord how is this? Sorry for stripping some spaces at the end of the lines.

**Comment by beoran** [comment]
If no one has any more comments I will merge this tomorrow.

**Comment by fatcerberus** [comment]
Looks good to me.  Static buffer is 255; nobody should need to set a window title longer than that. :-)

**Pull request merged**